ui/appui-abstract/src/appui-abstract/UiItemsProvider.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/* eslint-disable deprecation/deprecation */
+/** @packageDocumentation
+ * @module UiItemsProvider
+ */
+
+import { BackstageItem } from "./backstage/BackstageItem";
+import { CommonStatusBarItem } from "./statusbar/StatusBarItem";
+import { CommonToolbarItem, ToolbarOrientation, ToolbarUsage } from "./toolbars/ToolbarItem";
+import { AbstractWidgetProps } from "./widget/AbstractWidgetProps";
+import { AbstractZoneLocation, StagePanelLocation, StagePanelSection } from "./widget/StagePanel";
+
+/** Describes interface of objects that want to provide UI component to the running IModelApp.
+ * @public
+ */
+export interface UiItemsProvider {
+  /** id of provider */
+  readonly id: string;
+
+  /** UiItemsManager calls following method to get items to populate specific toolbars */
+  provideToolbarButtonItems?: (stageId: string, stageUsage: string, toolbarUsage: ToolbarUsage, toolbarOrientation: ToolbarOrientation, stageAppData?: any) => CommonToolbarItem[];
```

Comment:
we should talk later about deprecating these and providing an overload for these so you just pass in an object instead of a growing list of args

Reply:
Right. Let's target updating all these for 3.3